scripts/modules/helperFunctions.py

```diff
@@ -4,21 +4,64 @@
     remove_groundplane()
     extract_pcl_features()
 """
+import numpy as np
 
 GROUND_THRESHOLD = 1  # meters
 
 def remove_groundplane(pcl, z_thresh=GROUND_THRESHOLD):
     """Remove points below z-threshold and return pcl."""
     return pcl[pcl[:,2] > z_thresh]
 
-def extract_cluster_features(cluster):
-    """Calculate features for net training from pcl, return in some format.
+def extract_cluster_features(cluster, bbox):
+    """Calculate features for net training from pcl, return PclFeatures object
+    containing all features.
 
     Args:
-        cluster: ndarray (n * 3) of xyz points.
+        cluster: ndarray (n * 3) of cluster points.
+        bbox: waymo object label output
 
     Returns:
-        ???
+        features: Features object containing cluster features
 
     """
-    pass
+
+    id = bbox.id
+
+    # Compute eigenvalues along all three axes
+    xyz_cov = np.cov(np.transpose(cluster))
+    e_x, e_y, e_z = np.linalg.eigvals(xyz_cov)
+
+    # Compute volume and point density
+    vol = bbox.box.width * bbox.box.height
+    density = cluster.shape[0]/vol
+
+    output = Features(id, e_x, e_y, e_z, vol, density)
+    return output
+
+class Features(object):
```

Comment:
Good choice making a class. However, you could also do a namedTuple, which would be quite a bit smaller and cleaner, and work in basically the same way. I only say this because you're using this as, essentially, a custom dataset framework.